pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -151,5 +177,732 @@ func CreateCustomResourceDefinitions(clientset apiextensionsclient.Interface) er
+						"empty, all client identifiers are allowed.",
+					Type: "string",
+				},
+				"topic": {
+					Description: "Topic is the topic name contained in the message. If a Kafka " +
+						"request contains multiple topics, then all topics must be allowed or the " +
+						"message will be rejected.\n\nThis constraint is ignored if the matched " +
+						"request message type doesn't contain any topic. Maximum size of Topic can " +
+						"be 249 characters as per recent Kafka spec and allowed characters are " +
+						"a-z, A-Z, 0-9, -, . and _ Older Kafka versions had longer topic lengths " +
+						"of 255, but in Kafka 0.10 version the length was changed from 255 to 249. " +
+						"For compatibility reasons we are using 255\n\nIf omitted or empty, all " +
+						"topics are allowed.",
+					Type:      "string",
+					MaxLength: getInt64(255),
+				},
+			},
+		},
+		"Rule": {
+			Description: "Rule is a policy rule which must be applied to all endpoints which match " +
+				"the labels contained in the endpointSelector\n\nEach rule is split into an " +
+				"ingress section which contains all rules applicable at ingress, and an egress " +
+				"section applicable at egress. For rule types such as `L4Rule` and `CIDR` which " +
+				"can be applied at both ingress and egress, both ingress and egress side have to " +
+				"either specifically allow the connection or one side has to be omitted.\n\n" +
+				"Either ingress, egress, or both can be provided. If both ingress and egress are " +
+				"omitted, the rule has no effect.",
+			Required: []string{
+				"endpointSelector",
+			},
+			Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+				"Description": {
+					Description: "Description is a free form string, it can be used by the creator " +
+						"of the rule to store human readable explanation of the purpose of this " +
+						"rule. Rules cannot be identified by comment.",
+					Type: "string",
+				},
+				"egress": {
+					Description: "Egress is a list of EgressRule which are enforced at egress. If " +
+						"omitted or empty, this rule does not apply at egress.",
+					Type: "array",
+					Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+						Schema: &apiextensionsv1beta1.JSONSchemaProps{
+							Ref: getStr("#/properties/EgressRule"),
+						},
+					},
+				},
+				"endpointSelector": {
+					Description: "EndpointSelector selects all endpoints which should be subject " +
+						"to this rule. Cannot be empty.",
+					Ref: getStr("#/properties/EndpointSelector"),
+				},
+				"ingress": {
+					Description: "Ingress is a list of IngressRule which are enforced at ingress. " +
+						"If omitted or empty, this rule does not apply at ingress.",
+					Type: "array",
+					Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+						Schema: &apiextensionsv1beta1.JSONSchemaProps{
+							Ref: getStr("#/properties/IngressRule"),
+						},
+					},
+				},
```

Comment:
Do we need to make changes for `"egress" : ` too?

Reply:
with regarding?

Reply:
Yes. I see it now. Thanks !